src/renderer/components/BalanceGraph/BalanceGraph.tsx

```diff
@@ -0,0 +1,180 @@
+    //MONTH-last 12 days - 12 values
+    //YEAR-last 12 months- 12 values
+}
+
+export const BalanceGraph: React.FunctionComponent<IBalanceGraphProps> = (props: IBalanceGraphProps) => {
+    const [data, setData] = useState<Array<object>>([]);
+    const [interval, setInterval] = useState<IntervalEnum>(props.defaultInterval);
+    const [refreshInterval, setRefreshInterval] = useState<number>(0);
+    const [currentHour, setCurrentHour] = useState<number>(0);
+    const [currentDay, setCurrentDay] = useState<number>(0);
+    const [currentMonth, setCurrentMonth] = useState<number>(0);
+
+    const setXAxis = (array: number[], interval: IntervalEnum): void=>{
+        let dataArray: Array<object> = [];
+        switch (interval) {
+            case IntervalEnum.HOUR: dataArray=normalizeHour(dataArray,array);
+                break;
+            case IntervalEnum.DAY: dataArray=normalizeDay(dataArray,array);
+                break;
+            case IntervalEnum.WEEK: dataArray=normalizeWeek(dataArray,array);
+                break;
+            case IntervalEnum.MONTH: dataArray=normalizeMonth(dataArray,array);
+                break;
+            case IntervalEnum.YEAR: dataArray=normalizeYear(dataArray,array);
+                break;
+        }
+        setData(dataArray);
+    };
+
+    const handleOptionClick = (IntervalValue: IntervalEnum): void=>{
+        setInterval(IntervalValue);
+        props.getData(interval);
+    };
+
+    const renderGraphIntervalOption = (IntervalValue: IntervalEnum): string => {
+        let selector = "";
+        interval===IntervalValue ? selector="selected" : null;
+        return `graph-option ${selector}`;
+    };
+
+    const awaitData = async (): Promise<void>=>{
+        props.getData(interval).then((dataValueArray)=>{
+            setXAxis(dataValueArray, interval);
+        });
+    };
+
+    useEffect(()=>{
+        
+        const timeOnMount = new Date();
+        setCurrentHour(timeOnMount.getHours());
+        setCurrentDay(timeOnMount.getDay());
+        setCurrentMonth(timeOnMount.getMonth());
+
+        switch (interval) {
+            case IntervalEnum.HOUR: {
+                const localRefreshInterval = window.setInterval(awaitData, 60000);
+                setRefreshInterval(localRefreshInterval);
+            }
+                break;
+            case IntervalEnum.DAY: {
+                const intervalHandler = (): void =>{
+                    const timeOnInterval = new Date();
+                    if(currentHour===23){
+                        if(timeOnInterval.getHours()===0){
+                            setCurrentHour(timeOnInterval.getHours());
+                            awaitData();
+                        }
+                    } else {
+                        if(timeOnInterval.getHours()>currentHour){
+                            setCurrentHour(timeOnInterval.getHours());
+                            awaitData();
+                        }
+                    }
```

Comment:
You can make this code more generic by storing lastRefreshTime = new Date() in state
and on each interval do `const diffInSeconds = new Date().getTime()/ 1000 - lastRefreshTime.getTime()/ 1000`

and then do switch case:
```
switch(interval)
case Day:
if(diffInSeconds >= dayToSeconds) {
   refresh()
}
```